src/controllers/session.js

```diff
@@ -0,0 +1,68 @@
+const bcrypt = require('bcrypt')
+const jwt = require('jsonwebtoken')
+
+const Users = require('../models/User.js')
+const roles = require('../constants/roles.js')
+
+const sessionController = module.exports = {}
+
+sessionController.register = (req, res, next) => {
+  Users.find({ username: req.body.username }).then(user => {
+    if (user.length === 0) {
+      return bcrypt.hash(req.body.password, 12)
+    }
+
+    res.boom.badRequest('username already exists')
+  }).then(passwordHash => {
+    const { englishName, arabicName, email, username, imageUrl } = req.body
+    const en = englishName && { name: englishName }
+    const ar = arabicName && { name: arabicName }
+    return Users.create(Object.assign(
```

Comment:
we have just set them but it's about making sure that the new object has it's own things that are `ar` or `en` and don't point to other things in memory that are potentially mutatable.

I think. @des-des ?